pkg/cmd/issue/edit/edit.go

```diff
@@ -141,75 +149,107 @@ func editRun(opts *EditOptions) error {
 		return err
 	}
 
-	editable.Title.Default = issue.Title
-	editable.Body.Default = issue.Body
-	editable.Assignees.Default = issue.Assignees.Logins()
-	editable.Labels.Default = issue.Labels.Names()
-	editable.Projects.Default = append(issue.ProjectCards.ProjectNames(), issue.ProjectItems.ProjectTitles()...)
-	projectItems := map[string]string{}
-	for _, n := range issue.ProjectItems.Nodes {
-		projectItems[n.Project.ID] = n.ID
-	}
-	editable.Projects.ProjectItems = projectItems
-	if issue.Milestone != nil {
-		editable.Milestone.Default = issue.Milestone.Title
-	}
-
-	if opts.Interactive {
-		err = opts.FieldsToEditSurvey(&editable)
-		if err != nil {
-			return err
-		}
-	}
-
+	// Fetch editable shared fields once for all issues.
 	apiClient := api.NewClientFromHTTP(httpClient)
-	opts.IO.StartProgressIndicator()
+	opts.IO.StartProgressIndicatorWithLabel("Fetching repository information")
 	err = opts.FetchOptions(apiClient, repo, &editable)
 	opts.IO.StopProgressIndicator()
 	if err != nil {
 		return err
 	}
 
-	if opts.Interactive {
-		editorCommand, err := opts.DetermineEditor()
-		if err != nil {
-			return err
-		}
-		err = opts.EditFieldsSurvey(&editable, editorCommand)
-		if err != nil {
-			return err
-		}
+	// Update all issues in parallel.
+	issueURLs := make(chan string, len(issues))
+	if opts.workers == 0 {
+		opts.workers = 10
 	}
+	g := errgroup.Group{}
+	g.SetLimit(opts.workers)
+
+	opts.IO.StartProgressIndicatorWithLabel(fmt.Sprintf("Updating %d issues", len(issues)))
+	for i := range issues {
+		// Copy variables to capture in the go routine below.
+		issue := issues[i]
+		editable := editable.Clone()
+
+		g.Go(func() error {
+			editable.Title.Default = issue.Title
+			editable.Body.Default = issue.Body
+			editable.Assignees.Default = issue.Assignees.Logins()
+			editable.Labels.Default = issue.Labels.Names()
+			editable.Projects.Default = append(issue.ProjectCards.ProjectNames(), issue.ProjectItems.ProjectTitles()...)
+			projectItems := map[string]string{}
+			for _, n := range issue.ProjectItems.Nodes {
+				projectItems[n.Project.ID] = n.ID
+			}
+			editable.Projects.ProjectItems = projectItems
+			if issue.Milestone != nil {
+				editable.Milestone.Default = issue.Milestone.Title
+			}
+
+			// Allow interactive prompts for one issue; failed earlier if multiple issues specified.
+			if opts.Interactive {
+				editorCommand, err := opts.DetermineEditor()
+				if err != nil {
+					return err
+				}
+				err = opts.EditFieldsSurvey(&editable, editorCommand)
+				if err != nil {
+					return err
+				}
+			}
+
+			err = prShared.UpdateIssue(httpClient, repo, issue.ID, issue.IsPullRequest(), editable)
+			if err != nil {
+				return err
+			}
 
-	opts.IO.StartProgressIndicator()
-	err = prShared.UpdateIssue(httpClient, repo, issue.ID, issue.IsPullRequest(), editable)
+			issueURLs <- issue.URL
+			return nil
+		})
+	}
 	opts.IO.StopProgressIndicator()
+
+	err = g.Wait()
+	close(issueURLs)
+
 	if err != nil {
```

Comment:
As we now potentially have some issues being successfully edited and some not I think we alert the user which ones have been successful and which ones either failed or were not attempted. Right now if there is an error the user will just see the error and won't know if any of the issues were properly edited.

Reply:
Good point. I should still probably dump those that were successfully edited to stdout. `errgroup` will return on the first error, though because none of the API calls take a `context` I can't pass in a `context` that `errgroup.WithContext` would return, thus cancelling the rest. Thus, I wonder if this err check should just be a return of the `err` or at least moved to the bottom of the function.

Since any existing callers should expect either a single line to stdout (by default) or an exit code, though, do you anticipate any problems? Callers would have to pass in multiple issues, so any existing callers wouldn't do that. If they did, they should expect this new potential mix of stdout + exit code. Seems reasonable, but wanted your take.

Reply:
Yeah... this is getting a bit more complicated unfortunately due to the fact that the network calls do not take a `context`. I think we might need to rethink the strategy a bit. 

I am thinking we move away from using `sync.ErrGroup` and just go with `sync.WaitGroup` here. We will kick off updating each issue and allow each update to run to completion. We can keep track of the ones that were successful and the ones that failed. Then for the output we can indicate which ones successfully updated and which ones failed. If any of them failed then we can return a non-zero exit code.

I think that as long as we keep the same output for when there is only one issue specified then this won't be a breaking change. When a user specifies multiple issues to update then I think it is reasonable that they would expect multiple line output.

Let me know what you think.

Reply:
I think `sync.WaitGroup` makes sense but agree it's a good idea to consider passing a `context` to network calls. If/when that happens, this could be updated to an `errgroup.Group`. Worth a `TODO` comment? I'll leave it off for now because I suspect not.